web/libs/editor/src/mixins/Tool.js

```diff
@@ -93,6 +93,13 @@ const ToolMixin = types
 
       if (typeof self[fn] !== "undefined") self[fn].call(self, ev, args);
     },
+
+    shouldSkipInteractions(e) {
+      const isCtrlPressed = e.evt && (e.evt.metaKey || e.evt.ctrlKey);
+      const hasSelection = self.control.annotation.hasSelection;
+
+      return isCtrlPressed && !hasSelection;
```

Comment:
I may be misunderstanding something but based on the description if there is no selection we create a new region overtop existing regions. (this seems to allow it as coded)
but if there is a selection then we add to the selection - would that not go through here and fail? or does that go through another spot?

Reply:
>based on the description if there is no selection we create a new region overtop existing regions
> if there is a selection then we add to the selection
Only when the meta key is pressed, otherwise its the default behaviour as before which depends on the state as to the operation.

Reply:
If we are going to skip the interaction, the event will be processed inside the tool and it will draw region https://github.com/HumanSignal/label-studio/blob/03837fca3292b03fd9afd485cee2b842af77b945/web/libs/editor/src/components/ImageView/ImageView.jsx#L595-L617
With the currently selected region, we will skip this interaction, but clicking on the region will still be processed.
https://github.com/HumanSignal/label-studio/blob/03837fca3292b03fd9afd485cee2b842af77b945/web/libs/editor/src/regions/RectRegion.jsx#L514-L522
And it affects selection 
https://github.com/HumanSignal/label-studio/blob/03837fca3292b03fd9afd485cee2b842af77b945/web/libs/editor/src/mixins/KonvaRegion.js#L128-L135
So with ctrl/cmd key pressed and drawing tool selected it might draw over existed region if there is no selection and add/remove to/from selection if there is already selected regions.

I hope that I've mentioned here all meaningful spots, 
'cause it's really a bit complicated. 😅

Reply:
Well. It looks like I have found the problem with `skipInteractions` that resets flag before we can check it it `item.parent.getSkipInteractions()` inside onClick.
So I do not need `allowedDrawingThroughRegion` anymore...